src/Tribe/REST/V1/Endpoints/Single_Event.php

```diff
@@ -560,12 +560,27 @@ protected function prepare_postarr( WP_REST_Request $request ) {
 			'EventCurrencySymbol'   => tribe( 'cost-utils' )->parse_currency_symbol( $request['cost'] ),
 			'EventURL'              => filter_var( $request['website'], FILTER_SANITIZE_URL ),
 			// Taxonomies
-			'tax_input'             => array_filter( array(
-				$events_cat => Tribe__Terms::translate_terms_to_ids( $request['categories'], $events_cat ),
-				'post_tag'  => Tribe__Terms::translate_terms_to_ids( $request['tags'], 'post_tag' ),
-			) ),
+			'tax_input'             => array(),
 		);
 
+		// Check if categories is provided (allowing for empty array to remove categories).
+		if ( isset( $request['categories'] ) ) {
+			$postarr[ $events_cat ] = array();
+
+			if ( ! empty( $request['categories'] ) ) {
+				$postarr[ $events_cat ] = Tribe__Terms::translate_terms_to_ids( $request['categories'], $events_cat );
+			}
+		}
+
+		// Check if tags is provided (allowing for empty array to remove tags).
+		if ( isset( $request['tags'] ) ) {
+			$postarr['post_tag'] = array();
```

Comment:
Same question / suggestion as above.

Reply:
Replied on https://github.com/moderntribe/the-events-calendar/pull/2100#discussion_r204536623